tests/nat/agent/test_reasoning_agent.py

```
@@ -373,9 +377,11 @@ def mock_get_function(name: str):
 
     # The builder says we have no tools
     class FakeDeps:
-        functions = []
+        functions = set()
```

Ruff (0.12.2)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)